sympy/geometry/ellipse.py

```diff
@@ -1339,30 +1343,80 @@ class Circle(Ellipse):
     >>> c2.hradius, c2.vradius, c2.radius, c2.center
     (sqrt(2)/2, sqrt(2)/2, sqrt(2)/2, Point2D(1/2, 1/2))
 
+    >>> # a circle object returned from given equation and optional 'x' and 'y' parameters
+    >>> from sympy.abc import x, y, a, b
+    >>> Circle(x ** 2 + y ** 2 - 25)
+    Circle(Point2D(0, 0), 5)
+    >>> Circle(a ** 2 + b ** 2 - 25, x='a', y='b')
+    Circle(Point2D(0, 0), 5)
+
     """
 
     def __new__(cls, *args, **kwargs):
-        c, r = None, None
-        if len(args) == 3:
-            args = [Point(a, dim=2) for a in args]
-            if Point.is_collinear(*args):
-                raise GeometryError(
-                    "Cannot construct a circle from three collinear points")
-            from .polygon import Triangle
-            t = Triangle(*args)
-            c = t.circumcenter
-            r = t.circumradius
-        elif len(args) == 2:
-            # Assume (center, radius) pair
-            c = Point(args[0], dim=2)
-            r = sympify(args[1])
-
-        if not (c is None or r is None):
-            if r == 0:
-                return c
-            return GeometryEntity.__new__(cls, c, r, **kwargs)
-
-        raise GeometryError("Circle.__new__ received unknown arguments")
+
+        if len(args) == 1 and isinstance(args[0], Expr):
+            x = kwargs.get('x', 'x')
+            y = kwargs.get('y', 'y')
+            equation = args[0]
+
+            def find(x, equation):
```

Comment:
Let's move this to geometry/util.py but not import it into sympy namespace. Just import it into line and ellipse where it is needed.